examples/todo-mvc/src/state.rs

```diff
@@ -0,0 +1,86 @@
+use serde::{Deserialize, Serialize};
+
+use crate::{entry::Entry, filter::Filter};
+
+#[derive(Deserialize, Serialize)]
+pub struct State {
+    pub entries: Vec<Entry>,
+    pub filter: Filter,
+    pub value: String,
+    pub edit_value: String,
+}
+
+impl State {
+    pub fn total(&self) -> usize {
+        self.entries.len()
+    }
+
+    pub fn total_completed(&self) -> usize {
+        self.entries.iter().filter(|entry| Filter::Completed.fits(entry)).count()
+    }
+
+    pub fn is_all_completed(&self) -> bool {
+        let mut filtered_iter = self.entries.iter().filter(|entry| self.filter.fits(entry)).peekable();
+
+        match filtered_iter.peek() {
+            Some(_) => filtered_iter.all(|entry| entry.completed),
+            None => false,
+        }
+    }
+
+    pub fn clear_completed(&mut self) {
+        let entries = self.entries.drain(..).filter(|entry| Filter::Active.fits(entry)).collect();
+        self.entries = entries;
+    }
+
+    pub fn toggle(&mut self, idx: usize) {
+        let filter = self.filter;
+        let entry = self.entries.iter_mut().filter(|entry| filter.fits(entry)).nth(idx).unwrap();
+        entry.completed = !entry.completed;
```

Comment:
```suggestion
        if let Some(entry) = self.entries.iter_mut().filter(|entry| filter.fits(entry)).nth(idx) {
            entry.completed = !entry.completed;
        }
```

Though the unwrap should never fail, this is just another way to help avoid the potential panic due to some future refactoring. Thoughts?